h/api/schemas.py

```diff
@@ -226,9 +223,11 @@ def validate(self, data):
         new_appstruct['text'] = appstruct.pop('text', u'')
         new_appstruct['tags'] = appstruct.pop('tags', [])
 
+        permissions = appstruct.pop('permissions', None)
+
         # Replace the client's complex permissions object with a simple shared
         # boolean.
-        if appstruct.pop('permissions')['read'] == [new_appstruct['userid']]:
+        if permissions and permissions['read'] == [new_appstruct['userid']]:
```

Comment:
I'm not sure if we want to make the assumption that when the permissions field is missing that it means the annotation is public?